Sources/AccountSDKIOSWeb/Client/Client.swift

```diff
@@ -168,6 +177,7 @@ public class Client: CustomStringConvertible {
                     SchibstedAccountLogger.instance.error("Login flow error: \(String(describing: error))")
                     completion(.failure(.unexpectedError(message: "ASWebAuthenticationSession failed: \(String(describing: error))")))
                 }
+                self.isSessionInProgress = false
                 return
             }
             self.handleAuthenticationResponse(url: url, completion: completion)
```

Comment:
Would it be easier to follow along, if the `self.isSessionInProgress` is also set back to false right before the `handleAuthenticationResponse` . At this point we know we have received a response, so it should be safe to reset the boolean here, as opposed to do it two places within the `handleAuthenticationResponse`

Reply:
Not exactly, because inside handleAuthenticationResponse we check and compare state value stored in stateStorage with one from response. There is also logic for removing that value. Putting it before handleAuthenticationResponse can cause overriding stored state value and comparison failure (when the next session is triggered before comparison is made for its predecessor ).